src/tracer.js

```diff
@@ -0,0 +1,73 @@
+'use strict'
+
+import BasicSpan from './span'
+import {TextMapPropagator, BinaryPropagator} from './propagation'
+import {DefaultSampler} from './sampler'
+import {DefaultRecorder} from './recorder'
+
+// Implement https://github.com/opentracing/opentracing-javascript/blob/master/src/tracer.js
+export default class BasicTracer {
+    /**
+     * Constructs basic tracer instance with provided Sampler and Recorder
+     * @param  {Object} options
+     *         Optional associative array of fields.
+     *         - `sampler` {Sampler} Optional object with `isSample` method, the
+     *             method provided with current span as arguments, return
+     *             Boolean value indicate whether should take current span
+     *             as sample. See src/sample.js for example.
+     *         - `recorder` {Recorder} Optional object with `record` method, the
+     *             method take span and do whatever required to record a span.
+     *             See src/recorder.js for example.
+     *
+     * @return {BasicTracer}
+     */
+    constructor({sampler, recorder} = {}) {
+        this._sampler = sampler || new DefaultSampler()
+        this._recorder = recorder || new DefaultRecorder()
+        this._binaryPropagator = new BinaryPropagator(this)
+        this._textPropagator = new TextMapPropagator(this)
+    }
+
+    setInterface(inf) {
+        this._interface = inf
+    }
+
+    startSpan(fields) {
+        // Interface or Implementation argument
+        // https://github.com/opentracing/opentracing-javascript/pull/29
```

Comment:
I'm a little confused as well. I would have thought that `fields.parent` is _always_ a valid OpenTracing Span "Interface" object in the call to `startSpan`. Therefore, line 44 could simply be `parent: fields.parent.imp()` and line 39-41 could be removed. I'm not sure I understand how `Tracer` is involved.

My apologies if I'm missing something obvious here!